UI: change glyphs with actions and/or with on click to shy buttons with symbols

lukastocker opened the pull request:
- [x] Table > Data Table
- [x] Table > Presentation Table
- [x] Input > Password
- [x] Input > File
- [x] Main Controls > System Info
- [x] Entity > Standard

Also adjust PHPUnit tests and some examples accordingly.

lukastocker committed:
UI: data table: change rendering glyphs with actions to shy buttons with symbol.

lukastocker committed:
UI: password: change rendering glyphs with actions to shy buttons with symbol.

lukastocker committed:
UI: file input, change glyphs with actions to button with symbols.

lukastocker committed:
UI, entity: change rendering only glyphs for prioritized reactions to rendering buttons with symbols.

lukastocker committed:
UI: main controls > system info, change more and close glyph to shy buttons with symbols.

lukastocker committed:
UI: presentation table, change glyphs with actions to button with symbols and refactor rendering of presentation row.

nhaagen commented on components/ILIAS/UI/src/Implementation/Component/Entity/Renderer.php:
```diff
@@ -47,13 +47,11 @@ protected function renderEntity(Entity $component, RendererInterface $default_re
             $tpl->touchBlock('secondid_string');
-        } elseif ($secondary_identifier instanceof Component\Image\Image) {
-            $tpl->touchBlock('secondid_image');
-        } elseif ($secondary_identifier instanceof Component\Image\Symbol) {
+        } elseif ($secondary_identifier instanceof Component\Symbol\Symbol) {
             $tpl->touchBlock('secondid_symbol');
-        } elseif ($secondary_identifier instanceof Component\Image\Link) {
+        } elseif ($secondary_identifier instanceof Component\Link\Standard) {
             $tpl->touchBlock('secondid_link');
-        } elseif ($secondary_identifier instanceof Component\Image\Shy) {
+        } elseif ($secondary_identifier instanceof Component\Button\Shy) {
             $tpl->touchBlock('secondid_shy');
         }
```
I agree that the former code was quite strange, i.e. there is no Component\Image\Link or Component\Image\Symbol, but nonetheless will ::getSecondaryIdentifier return  Symbol | Image | Shy | StandardLink | string

nhaagen commented on components/ILIAS/UI/src/Implementation/Component/MainControls/Renderer.php:
```diff
@@ -320,15 +321,14 @@ protected function renderSystemInfo(
         }
 
-        $more = $this->getUIFactory()->symbol()->glyph()->more("#");
+        $more = $f->button()->shy('', '')->withSymbol($f->symbol()->glyph()->more());
```
` $more = $f->button()->shy('', '#')->withSymbol($f->symbol()->glyph()->more());`

nhaagen commented on components/ILIAS/UI/src/templates/default/Table/tpl.datatable.html:
```diff
@@ -22,7 +22,6 @@ <h2 class="ilHeader" id="{ID}_label">{TITLE}</h2>
 						<div class="c-table-data__header__resize-wrapper">
 							<!-- BEGIN header_cell_sortglyph -->
-							{COL_SORTATION_GLYPH}
 							<!-- END header_cell_sortglyph -->
```
you might drop the block as well, here.

nhaagen commented:
LGTM this far, thank you. 
Now please remove the deprecated methods on Glyph.
Please also find a solution regarding  ::withUnavailableAction() (same method is on buttons)  to visually disable the Glyph; I guess, this can be done by CSS.

lukastocker committed:
UI: disable glyph via SCSS if button with symbol is disabled...

...and remove deprecated functions from glyph like 'withAction' and 'withUnavailableAction' from the glyph
renderer, interface etc.

Adjust PHPUnit tests accordingly.

lukastocker commented:
- [x] Removed deprecated methods on Glyph
- [x] Adjusted PHPUnit Tests
- [x] withUnavailableAction for Glyphs is now handled by SCSS

nhaagen commented on components/ILIAS/UI/src/examples/Symbol/Glyph/Sortation/sortation.php:
```diff
@@ -22,8 +22,17 @@
- * expected output: >
- *   ILIAS shows the rendered Component.
+ *  description: >
+ *    Example for rendring a sortation glyph.
```
```suggestion
 *    Example for rendering a sortation glyph.
```

lukastocker committed:
UI: refactor glyphs with actions etc. to buttons with symbols and actions etc. in components/ILIAS
